java/opendcs/src/main/java/decodes/tsdb/algo/WeightedAverage.java

```diff
@@ -0,0 +1,202 @@
+        String t = null;
+        if (((pr = getParmRef(t = "input1")) != null && isAssigned(t) && isMissing(input1) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "input2")) != null && isAssigned(t) && isMissing(input2) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "input3")) != null && isAssigned(t) && isMissing(input3) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "input4")) != null && isAssigned(t) && isMissing(input4) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "input5")) != null && isAssigned(t) && isMissing(input5) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "input6")) != null && isAssigned(t) && isMissing(input6) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "input7")) != null && isAssigned(t) && isMissing(input7) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "input8")) != null && isAssigned(t) && isMissing(input8) &&
+              pr.missingAction != MissingAction.IGNORE))
+        {
+            log.debug("Skipping time slice with base time {} because of missing value for param {}",
+                    debugSdf.format(_timeSliceBaseTime), t);
+            return;
+        }
+        if (((pr = getParmRef(t = "weight1")) != null && isAssigned(t) && isMissing(weight1) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "weight2")) != null && isAssigned(t) && isMissing(weight2) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "weight3")) != null && isAssigned(t) && isMissing(weight3) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "weight4")) != null && isAssigned(t) && isMissing(weight4) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "weight5")) != null && isAssigned(t) && isMissing(weight5) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "weight6")) != null && isAssigned(t) && isMissing(weight6) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "weight7")) != null && isAssigned(t) && isMissing(weight7) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "weight8")) != null && isAssigned(t) && isMissing(weight8) &&
+              pr.missingAction != MissingAction.IGNORE))
+        {
+            log.debug("Skipping time slice with base time {} because of missing value for param {}",
+                    debugSdf.format(_timeSliceBaseTime), t);
+            return;
+        }
+        if (!isMissing(input1) && !isMissing(weight1))
+            tot += (input1 * weight1);
```

Comment:
Brackets. yeah i know it's an extra two lines and characters. I've been bit by this syntax too many times